woocommerce/woocommerce-android
# Disable custom amount

Closes: #12299

### Description
This PR refactors the `isLocked` logic of custom amounts to match the web behavior.  Now the custom amount section will follow the `isEditable` state and disable the editable mode if the `isLocked` value is true. I also refactored the `isLocked` field and move it from `CustomAmountUIModel` to `OrderCreateEditCustomAmountAdapter` and grouped the logic for enable/disable the editable mode to the functions `showEditableControls` and `hideEditableControls`.

### Steps to reproduce

1. Go to the Orders tab.
2. Select an order with the processing status, and without a custom amount.
3. Edit the order.
4. In the order form, add a custom amount. Notice that you can add a new amount even though the section has a lock icon.
5. In the order form, notice that the added custom amount can be edited.

#### Expected behavior

I expect that for orders with limited editing (e.g. with processing status):

1. Adding custom amounts will be disabled.
2. Editing custom amounts will be disabled.
3. When the section has editing disabled, the UI to add/edit the custom amounts will be hidden or show a disabled status.

### Testing information

1. Open the orders tab
2. Add a new order with products
3. Open the order created and check that when the order's status is not editable (e.g., processing), add/edit custom amount is disabled, and if the order is in an editable status (e.g., pending payment), add/edit custom amount is allowed.

### The tests that have been performed

1. Creating a new order without a custom amount and checked that when the status is `pending payment` a custom amount can be added to the order, and when the status is `processing` the add custom amount button is disabled.
2. Creating a new order with a custom amount and checked that when the status is `pending payment` a custom amount can be edited, and when the status is `processing` the custom amount is disabled and CAN'T be edited.

### Images/gif

https://github.com/user-attachments/assets/1400e81b-7137-45de-a2c5-f748dc6ac5ce


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR or request changes if the PR needs improvement:
